gamemodes/terrortown/gamemode/client/cl_search.lua

```diff
@@ -151,15 +151,20 @@ end
 -- @param string name A unique name to keep a reference to the item
 -- @param table data The data that is added to the element
 -- @param[opt] number height The height of the element
+-- @param[opt] number order The order id of the element in the list
 -- @return Panel The box element
 -- @realm client
-function SEARCHSCREEN:MakeInfoItem(parent, name, data, height)
+function SEARCHSCREEN:MakeInfoItem(parent, name, data, height, order)
     local box = vgui.Create("DInfoItemTTT2", parent)
     box:SetSize(self.sizes.widthContentBox, height or self.sizes.heightInfoItem)
     box:Dock(TOP)
     box:DockMargin(0, 0, 0, self.sizes.padding)
     box:SetData(data)
 
+    if isnumber(order) then
+        box:SetZPos(order)
```

Comment:
I think I don't understand why you have to set the ZPos here?

Reply:
When using Dock, ZPos changes the order of the docked panels. By default, each panel has a ZPos of 0 and is displayed in the order they were added.

Reply:
oh wow, I did not know that it works that way. I always used "manual" sorting for this